program/src/balance.rs

```diff
@@ -219,40 +228,44 @@ mod test {
             (0, Lamports(0))
         );
     }
+
     #[test]
-    fn get_target_balance_weighted_validators() {
-        let mut validators = Validators::new_fill_default(2);
-        validators.entries[0].entry.weight = Weight(1000);
-        validators.entries[1].entry.weight = Weight(3000);
+    fn get_target_balance_inactive_validator() {
+        // 200 Lamports delegated, but only one active validator,
+        // so all of the target should be with that one validator.
+        let mut validators = Validators::new_fill_default(3);
+        validators.entries[0].entry.stake_accounts_balance = Lamports(100);
+        validators.entries[1].entry.stake_accounts_balance = Lamports(0);
+        validators.entries[1].entry.inactive = true;
+        validators.entries[2].entry.stake_accounts_balance = Lamports(300);
 
-        let mut targets = [Lamports(0); 2];
-        let undelegated_stake = Lamports(100);
+        let mut targets = [Lamports(0); 3];
+        let undelegated_stake = Lamports(0);
         let result = get_target_balance(undelegated_stake, &validators, &mut targets[..]);
         assert!(result.is_ok());
-        assert_eq!(targets, [Lamports(25), Lamports(75)]);
+        assert_eq!(targets, [Lamports(200), Lamports(0), Lamports(200)]);
 
         assert_eq!(
             get_validator_furthest_below_target(&validators, &targets[..]),
-            (1, Lamports(75))
+            (0, Lamports(100))
         );
     }
-    #[test]
-    fn get_target_balance_weighted_validators_non_integer_multiple() {
```

Comment:
We should keep the non-integer multiple test, it was testing the distribution of the additional lamports at the end. In the new test above, with 400 lamports and 2 active validators, the total is a multiple of the number of validators, so there is no remainder to divide. We should also test the case where the total is not a multiple, and the target of some validators is therefore larger than that of others.

It should be possible to see this in the code coverage results too. You can run `tests/coverage.py` locally, and that will produce an html report where you can see which lines of the distribution logic are covered.

Reply:
Oops... will do that, thanks!

Reply:
Wouldn't this test be equal to the `get_target_balance_works_for_non_integer_multiple`?

Reply:
There is a small difference: when one of the validators is inactive, distributing the remainder should skip over those. To test it properly, we need to make sure then that the inactive validator(s) are early in the list, such that there is a validator behind an inactive one, that gets one additional lamport.